grpclog/component.go

```diff
@@ -0,0 +1,253 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpclog
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+
+	"google.golang.org/grpc/internal/grpclog"
+)
+
+// componentData records the settings for a component.
+type componentData struct {
+	name      string
+	verbosity int
+	level     int
+}
+
+const envName = "GRPC_GO_LOG_LEVEL"
+const sentinel = math.MinInt32
+
+// log levels
+const (
+	levelInfo    = 0
+	levelWarning = 1
+	levelError   = 2
+)
+
+var environmentVars = map[string]*componentData{}
+var prefixVars = map[string]*componentData{}
+var cache = map[string]*componentData{}
+
+// init creates all grpc components and any components named in the environment
```

Comment:
IIUC this doesn't create any components now, right?

Reply:
You are right, updated the comment.